kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/Contributor.java

```diff
@@ -16,13 +16,30 @@
 public interface Contributor<T, S extends Context> {
 
     /**
+     * Identifies if this Contributor instance can create instances for the provided short name.
+     * @param shortName The type name to test for.
+     * @return <code>true</code> if the <code>shortName</code> is supported by this contributor.
+     */
+    boolean contributes(String shortName);
```

Comment:
While we're here, Tom asked what a shortName is. Should we take the chance to rename it to `name` or `id` or something like that? Maybe `type` to match the YAML naming?

Reply:
Yeah, I have been moving to `typeName` but not in a through fashion

Reply:
cool, the rename won't break binary compat will it, so we could leave it and do it any time

Reply:
Yeah renames are binary but not source compatible.

Reply:
I'm switching to type name where I touch it.